File: 83-remove-duplicates-from-sorted-list/remove-duplicates-from-sorted-list.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution(object):
    def deleteDuplicates(self, head):
        """
        :type head: Optional[ListNode]
        :rtype: Optional[ListNode]
        """
        if(not head):
            return None

        left=head
        right=head.next
        while(right):
            if(left.val!=right.val):
                left.next=right
                left=left.next
            right=right.next

        left.next=None
        return head
            

        

File: 83-remove-duplicates-from-sorted-list/remove-duplicates-from-sorted-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode deleteDuplicates(ListNode head) {
        if(head==null){
            return null;
        }
        ListNode cur=head;
        ListNode nextNode=head.next;
        while(cur!=null && nextNode!=null){
            if(cur.val==nextNode.val){
                cur.next=nextNode.next;
                nextNode=nextNode.next;
            }else{
                cur=nextNode;
                nextNode=nextNode.next;
            }
        }
        return head;
    }
}